lib/ansible/modules/cloud/amazon/cloudtrail.py

```diff
@@ -492,6 +485,9 @@ def main():
     if module.params['cloudwatch_logs_log_group_arn']:
         ct_params['CloudWatchLogsLogGroupArn'] = module.params['cloudwatch_logs_log_group_arn']
 
+    if module.params['enable_log_file_validation']:
+        ct_params['EnableLogFileValidation'] = module.params['enable_log_file_validation']
```

Comment:
Hmm. The problem with this is that it isn't possible to turn off log file validation, making this fail the idempotency test (you can't assert that log file validation is disabled). I'm not sure if the 'bool' param allows for the param to be [omitted](http://docs.ansible.com/ansible/latest/playbooks_filters.html#omitting-parameters) or if it defaults to false; if you can't test that, I'll try to get to it in the next few days/weeks.

Reply:
Huh... I don't think I totally follow, sorry. If you tried to assert it was disabled, anything getting info from 'get-trail-status' would report that it is disabled whether it's set to false or not specified, because "disabled" is the default of the service itself for new trails.

Using default(omit) in a task doesn't remove module defaults, I don't think - just only passes task params if they're set. Happy to test tomorrow UK time.

Reply:
Here's the use case:

- enable cloudtrail, set log file validation=true
- set log file validation=false

Given the conditional in 488, validation would still be enabled.

Reply:
I just tested this and can confirm that once you enable log file validation you can't set it back to False due to the conditional on 488.

`AnsibleModule` by default sets parameters to None if they are not specified and do not have a Default set, regardless of their type. So in the current code I think the above behavior can be avoided by changing line 488 to:
```python
if module.params['enable_log_file_validation'] is not None:
    ct_params['EnableLogFileValidation'] = module.params['enable_log_file_validation']
```

Reply:
D'oh - yup, that makes perfect sense by the light of day. Will update + fix the check mode.

Reply:
No worries. Thanks for your patience.